java/compaction-core/src/main/java/sleeper/compaction/job/status/CompactionJobStatus.java

```diff
@@ -138,22 +138,48 @@ private Instant firstTime() {
 
     public Instant lastTime() {
         if (isFinished()) {
-            return finishedStatus.getUpdateTime();
+            return getLatestFinishedStatus().getUpdateTime();
         } else if (isStarted()) {
-            return startedStatus.getUpdateTime();
+            return getLatestStartedStatus().getUpdateTime();
         } else {
             return createdStatus.getUpdateTime();
         }
     }
 
+    public CompactionJobStartedStatus getLatestStartedStatus() {
+        return jobRunList.stream()
+                .map(CompactionJobRun::getStartedStatus)
+                .max(Comparator.comparing(CompactionJobStartedStatus::getUpdateTime))
+                .orElse(null);
+    }
+
+    public CompactionJobFinishedStatus getLatestFinishedStatus() {
+        return jobRunList.stream()
+                .filter(CompactionJobRun::isFinished)
+                .map(CompactionJobRun::getFinishedStatus)
+                .max(Comparator.comparing(CompactionJobFinishedStatus::getUpdateTime))
+                .orElse(null);
+    }
+
+    public CompactionJobRun getLatestJobRun() {
+        return jobRunList.stream()
+                .max(Comparator.comparing(CompactionJobRun::getLatestUpdateTime))
+                .orElse(null);
+    }
```

Comment:
These seem to get called a lot, we could sort the list in the constructor?

Reply:
It's relying on the sort order set in CompactionJobStatusesBuilder now. We could maybe use naming to make it clear it needs to be sorted before you build CompactionJobStatus, something like jobRunsLatestFirst?